sql/expression/function/time_test.go

```diff
@@ -92,6 +92,127 @@ func TestTime_Month(t *testing.T) {
 	}
 }
 
+func TestTime_Quarter(t *testing.T) {
+	ctx := sql.NewEmptyContext()
+	f := NewQuarter(expression.NewGetField(0, types.LongText, "foo", false))
+
+	testCases := []struct {
+		name     string
+		row      sql.Row
+		expected interface{}
+		err      bool
+	}{
+		{
+			name:     "null date",
+			row:      sql.NewRow(nil),
+			expected: nil,
+		},
+		{
+			name:     "1",
+			row:      sql.NewRow(1),
+			expected: int32(1),
+		},
+		{
+			name:     "1.1",
+			row:      sql.NewRow(1.1),
+			expected: int32(1),
+		},
+		{
+			name:     "invalid type",
+			row:      sql.NewRow([]byte{0, 1, 2}),
+			expected: int32(1),
+		},
```

Comment:
Seems like all of these should return `NULL`, right? 

Here's the MySQL behavior:
```sql
mysql> select quarter("1"), quarter("1.1");
+--------------+----------------+
| quarter("1") | quarter("1.1") |
+--------------+----------------+
|         NULL |           NULL |
+--------------+----------------+
```